zetaclient/config/types.go

```diff
@@ -165,34 +159,11 @@ func (c Config) GetKeyringBackend() KeyringBackend {
 	return c.KeyringBackend
 }
 
-// LoadSolanaPrivateKey loads the Solana private key from the key file
-func (c Config) LoadSolanaPrivateKey() (solana.PrivateKey, error) {
-	// key file path
-	fileName := path.Join(c.ZetaCoreHome, c.SolanaKeyFile)
-
-	// load the gateway keypair from a JSON file
-	// #nosec G304 -- user is allowed to specify the key file
-	fileContent, err := os.ReadFile(fileName)
-	if err != nil {
-		return solana.PrivateKey{}, errors.Wrapf(err, "unable to read Solana key file: %s", fileName)
-	}
-
-	// unmarshal the JSON content into a slice of bytes
-	var keyBytes []byte
-	err = json.Unmarshal(fileContent, &keyBytes)
-	if err != nil {
-		return solana.PrivateKey{}, errors.Wrap(err, "unable to unmarshal Solana key bytes")
-	}
-
-	// ensure the key length is 64 bytes
-	if len(keyBytes) != 64 {
-		return solana.PrivateKey{}, fmt.Errorf("invalid Solana key length: %d", len(keyBytes))
-	}
-
-	// create private key from the key bytes
-	privKey := solana.PrivateKey(keyBytes)
-
-	return privKey, nil
+// GetRelayerKeyPath returns the relayer key path
+func (c Config) GetRelayerKeyPath() string {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	return c.RelayerKeyPath
```

Comment:
Do we really need a mutex here? It seems that we generate the `Config` when we initialize the client, so it shouldn't be needed as it's not going to change.

Is there any other code path where `RelayerKeyPath` changes at runtime?

Reply:
good point, it is static and never change.

Reply:
I just noticed that the config file could change dynamically. An example is:

The operator could modify the file by adding a newly integrated chain's endpoint and `zetaclientd` will then automatically pick up the new config file to activate the new chain.